lib/avromatic/io/datum_writer.rb

```diff
@@ -20,6 +20,14 @@ def write_union(writers_schema, datum, encoder)
         encoder.write_long(index_of_schema)
         write_data(writers_schema.schemas[index_of_schema], datum, encoder)
       end
+
+      def write_record(writers_schema, datum, encoder)
+        if datum.is_a?(Hash) && datum.key?(Avromatic::IO::ENCODING_PROVIDER)
+          encoder.write(datum[Avromatic::IO::ENCODING_PROVIDER].avro_raw_value)
+        else
+          super
+        end
+      end
```

Comment:
I think the other approach here would be to avoid the hash and just check for an Avromatic model:

```ruby
def write_record(writers_schema, datum, encoder)
  if value.is_a?(Avromatic::Model::Attributes)
    encoder.write(datum.avro_raw_value)
  else
    super
  end
end
```